### PR TITLE
MOTECH-1960: Fixed SubscriptionServiceImplTest

### PR DESCRIPTION
**Pull request by tstalka**
Now subscribe method returns id of the IntentVerificationThreadRunnable. If thread wasn't created then null is returned.

### CODE REVIEW

File: docs/source/hub.rst [hunk 1]
```diff
@@ -132,13 +132,14 @@ The Hub module exposes OSGi ``SubscriptionService`` for subscribing:
...
-        void subscribe(String callbackUrl, Modes hubMode, String topic,
+        Long subscribe(String callbackUrl, Modes hubMode, String topic,
                 String leaseSeconds, String secret) throws HubException;
 
     }
 
-To subscribe with this service just call subscribe method passing all the parameters. Detailed description of the parameters
-is shown in table below.
+To subscribe with this service just call subscribe method passing all the parameters. This method returns id of the thread
+which will verify the intent of the subscriber requesting subscription. In case of hubMode is ``unsubscribe`` thread isn't
```
Comment (pgesek) [inline docs/source/hub.rst]: if hubMode 
the thread won't be created and the method will return null

File: hub/src/main/java/org/motechproject/hub/service/SubscriptionService.java [hunk 2]
```diff
@@ -15,7 +15,9 @@
     /**
      * This method executes the business logic for subscribing/unsubscribing to
      * a topic. The subscriber provides the callback URL where the update
-     * notification should be sent.
+     * notification should be sent. Returns the id of the IntentVerificationThreadRunnable
+     * thread which verify the intent of the subscriber requesting subscription. In case of hubMode
```
Comment (pgesek) [inline hub/src/main/java/org/motechproject/hub/service/SubscriptionService.java]: verifies

Comment (pgesek) [inline hub/src/main/java/org/motechproject/hub/service/SubscriptionService.java]: if hubMode

File: hub/src/main/java/org/motechproject/hub/service/SubscriptionService.java [hunk 3]
```diff
@@ -15,7 +15,9 @@
...
+     * is "unsubscribe" thread isn't created and method returns null.
```
Comment (pgesek) [inline hub/src/main/java/org/motechproject/hub/service/SubscriptionService.java]: the thread won't be created and the method will return null

File: hub/src/main/java/org/motechproject/hub/service/SubscriptionService.java [hunk 4]
```diff
@@ -34,11 +36,12 @@
      *            - a <code>String</code> provided by the subscriber which will
      *            be used to compute an HMAC digest for authorized content
      *            distribution. Currently this is not being consumed by the API
+     * @return the id of IntentVerificationThreadRunnable which is created, if thread wasn't created then returns null
```
Comment (pgesek) [inline hub/src/main/java/org/motechproject/hub/service/SubscriptionService.java]: if the thread

**Comment by tstalka** [comment]
Test this please

**Comment by tstalka** [comment]
test this please

**Comment by pgesek** [comment]
test this please

**Comment by tstalka** [comment]
test this please

**Pull request merged**